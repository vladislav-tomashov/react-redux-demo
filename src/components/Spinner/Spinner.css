.Spinner {
  height: 100vh;
  width: 100%;
  position: absolute;
  align-items: center;
  display: flex;
  justify-content: center;
  background-color: rgba(150, 150, 150, 0.8);
}

.Spinner-image {
  animation: Spinner-image-spin infinite 1s linear;
  height: 20vmin;
}

@keyframes Spinner-image-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
