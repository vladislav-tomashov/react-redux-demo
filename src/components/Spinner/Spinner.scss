@import "../../styles/settings";

.Spinner {
  height: 100vh;
  width: 100%;
  position: absolute;
  align-items: center;
  display: flex;
  justify-content: center;
  background-color: rgba($black, 0.6);
}

.Spinner-image {
  animation: Spinner-image-spin infinite 1s linear;
  height: 20vmin;
}

@keyframes Spinner-image-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
