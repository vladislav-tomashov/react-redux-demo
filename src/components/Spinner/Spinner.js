import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { isCurrencyRatesLoading } from "../../store/selectors/currencyRatesSelectors";
import { ReactComponent as Logo } from "./logo.svg";
import "./Spinner.scss";

const Spinner = ({ loading = false } = {}) => {
  if (!loading) {
    return null;
  }
  return (
    <div className="Spinner">
      <Logo className="Spinner-image" />
    </div>
  );
};

Spinner.propTypes = {
  loading: PropTypes.bool
};

const mapStateToProps = state => {
  return {
    loading: isCurrencyRatesLoading(state)
  };
};

const ConnectedSpinner = connect(mapStateToProps)(Spinner);

export { ConnectedSpinner as default, Spinner };
