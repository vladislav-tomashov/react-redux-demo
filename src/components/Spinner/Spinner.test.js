import React from "react";
import { shallow } from "enzyme";
import { Spinner } from "./Spinner";

let wrapper;

describe("Spinner component", () => {
  beforeEach(() => {
    wrapper = shallow(<Spinner />);
  });

  test("should render Spinner correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render Spinner with 'loading' props correctly", () => {
    wrapper.setProps({
      loading: true
    });
    expect(wrapper).toMatchSnapshot();
  });
});
