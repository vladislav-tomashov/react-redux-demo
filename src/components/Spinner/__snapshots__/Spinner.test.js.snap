// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should render Spinner correctly 1`] = `""`;

exports[`Spinner component should render Spinner with 'loading' props correctly 1`] = `
<div
  className="Spinner"
>
  <ReactComponent
    className="Spinner-image"
  />
</div>
`;
